client/main.lua

```diff
@@ -260,17 +255,114 @@ RegisterNetEvent('qbx_weed:client:foodPlant', function()
             },
         })
     then
-        ClearPedTasks(cache.ped)
         local newFood = math.random(40, 60)
-        TriggerServerEvent('qbx_weed:server:foodPlant', currentHouse, newFood, data.sort, data.id)
+        TriggerServerEvent('qbx_weed:server:foodPlant', currentProperty, newFood, data.sort, data.id)
     else
-        ClearPedTasks(cache.ped)
-        LocalPlayer.state:set('invBusy', false, true)
         exports.qbx_core:Notify(locale('error.process_canceled'), 'error')
     end
+
+    ClearPedTasks(cache.ped)
+end)
+
+RegisterNetEvent('qbx_core:client:onSetMetaData', function(meta, oldValue, value)
+    if meta ~= 'currentPropertyId' then return end
+
+    currentProperty = value and tostring(value) or nil
+
+    if currentProperty then
+        despawnOutsidePlants(oldValue)
+
+        Wait(1000)
+
+        outsidePlants = {}
+
+        local plants = lib.callback.await('qbx_weed:server:getPropertyPlants', false, currentProperty)
+        propertyPlants[currentProperty] = plants
+        spawnPropertyPlants()
+    else
+        despawnPropertyPlants(oldValue)
+
+        Wait(1000)
+
+        propertyPlants[oldValue] = nil
+
+        Wait(2000) -- Wait for server to send outsidePlants
```

Comment:
I'm having a hard time understanding this Wait/comment. Where is the callback or net event that this comment is referring to?

Reply:
It's referring to the event handler on the server onMetadata that calls refreshOutsidePlants

Reply:
It's too long of a wait, but that's on purpose just in case it can take a little longer